torch/csrc/autograd/python_cpp_function.cpp

```diff
@@ -78,7 +78,10 @@ int THPCppFunction_traverse(PyObject* self, visitproc visit, void *arg)
 
 int THPCppFunction_clear(PyObject* self)
 {
-  ((THPCppFunction*)self)->cdata.reset();
+  auto f = (THPCppFunction*)self;
+  // Remove the weak ref of the c++ object if it exist
+  f->cdata->pyobj = nullptr;
```

Comment:
I think you should also guard this with an if-statement, e.g.:

```
if (f->cdata) {
  f->cdata->pyobj = nullptr;
}
```

I think there are cases where `tp_clear` can be called multiple times. For example, if an object's destructor "resurrects" a "dead" object by creating a new reference to it. We saw this happening in some other class (I forget which one).